overpass/api.py

```
@@ -191,21 +192,49 @@ def _as_geojson(self, elements):
-                if geom:
-                    for coords in elem.get("geometry"):
-                        points.append((coords["lon"], coords["lat"]))
-                    geometry = geojson.LineString(points)
+            elif elem_type == "way":
+                # Create LineString geometry
+                geometry = geojson.LineString([(coords["lon"], coords["lat"]) for coords in elem_geom])
+            elif elem_type == "relation":
+                # Initialize polygon list
+                polygons = []
+                # First obtain the outer polygons
+                for member in elem.get("members", []):
+                    if member["role"] == "outer":
+                        points = [(coords["lon"], coords["lat"]) for coords in member.get('geometry', [])]
+                        # Check that the outer polygon is complete
+                        if points and points[-1] == points[0]:
+                            polygons.append([points])
+                # Then get the inner polygons
+                for member in elem.get("members", []):
+                    if member["role"] == "inner":
+                        points = [(coords["lon"], coords["lat"]) for coords in member.get('geometry', [])]
+                        # Check that the inner polygon is complete
+                        if points and points[-1] == points[0]:
+                            # We need to check to which outer polygon the inner polygon belongs
+                            point = Point(points[0])
+                            for poly in polygons:
+                                polygon = Polygon(poly[0])
+                                if polygon.contains(point):
+                                    poly.append(points)
+                                    break
+                # Finally create MultiPolygon geometry
+                if polygons:
+                    geometry = geojson.MultiPolygon(polygons)
             else:
                 continue
```

Not changed by you but as OSM only supports the above three, the else clause should raise an exception.